src/aco/ants_system_v2/builder.rs

```diff
@@ -0,0 +1,69 @@
+use crate::aco::probe::Probe;
+use crate::aco::{AntSystem, AntSystemCfg, FMatrix};
+
+pub struct Builder {
+    conf: AntSystemCfg,
+}
+
+type Float = f64;
+
+impl Builder {
+    pub fn new() -> Self {
+        Builder {
+            conf: AntSystemCfg::default(),
+        }
+    }
+
+    pub fn set_weights(mut self, weights: FMatrix) -> Self {
+        self.conf.weights = weights;
+        self
+    }
+
+    pub fn set_heuristic(mut self, heuristic: FMatrix) -> Self {
+        self.conf.heuristic = heuristic;
+        self
+    }
+
+    pub fn set_alpha(mut self, alpha: Float) -> Self {
+        self.conf.alpha = alpha;
+        self
+    }
+
+    pub fn set_beta(mut self, beta: Float) -> Self {
+        self.conf.beta = beta;
+        self
+    }
+
+    pub fn set_evaporation_rate(mut self, evaporation_rate: Float) -> Self {
+        assert!(
+            evaporation_rate > 1 as Float || evaporation_rate < 0 as Float,
```

Comment:
You do not need casting here. Just write: 

```suggestion
            evaporation_rate > 1.0 || evaporation_rate < 0.0,
```